### PR TITLE
libbeat/processors/util: make mac address rendering conform to ECS

### LINKED ISSUE #32264: libbeat/processors/util: GetNetInfo formats hardware addresses incorrectly.
`GetNetInfo` currently formats hardware addresses using the standard library's String method for the `net.HardwareAddr` type (lower case hex octets separated by colons) which does not conform the ECS spec ([upper case hex octets separated by dashes](https://www.elastic.co/guide/en/ecs/current/ecs-host.html#field-host-mac)).

### PR DESCRIPTION
**Pull request by efd6**
## What does this PR do?

This alters the syntax of hardware addresses to match the syntax that is [specified in ECS](https://www.elastic.co/guide/en/ecs/current/ecs-host.html#field-host-mac).

## Why is it important?

Currently it does not match the spec and this leaks out to integrations and modules in ways that are irritating to fix in ingest pipelines.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] ~~I have made corresponding changes to the documentation~~
- [ ] ~~I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

- Closes #32264

### CODE REVIEW

File: libbeat/processors/util/netinfo.go [hunk 1]
```diff
@@ -67,6 +65,19 @@ func GetNetInfo() (ipList []string, hwList []string, err error) {
 	return ipList, unique(hwList), errs.Err()
 }
 
+// formatHardwareAddr formats hardware addresses according to the ECS spec.
+func formatHardwareAddr(addr net.HardwareAddr) string {
+	buf := make([]byte, 0, len(addr)*3-1)
```
Comment (cmacknz) [inline libbeat/processors/util/netinfo.go:70]: This will panic if passed `net.HardwareAddr{}`, can we guard against this?

Comment (efd6) [reply libbeat/processors/util/netinfo.go:70]: This is never called with an empty `net.HardwareAddr`.

**Comment by efd6** [comment]
I have the same concerns about the breaking change. The impetus for this is that the ECS specifies the format used in this change.

**Review by andrewkroh — APPROVED** [review]
> I suppose the argument for doing this is that anyone searching for MACs today would need to handle multiple possible formats already anyway since we don't standardize them.

Agree. I'm viewing this as a bug in that we are inconsistent with the format specified by ECS for MACs. Plus with all of the Fleet integrations using the ECS specified format, we often end up with a mix of formats in the same event.

There are workarounds if someone needs the non-ECS format. A runtime field could be used to shadow the value and produce both formats or just the non-ECS format. Or a beat processor or ingest node processor could be used to revert to the old format.

**Pull request merged**